services/collectd/service.go

```diff
@@ -385,6 +390,53 @@ func (s *Service) writePoints() {
 	}
 }
 
+// UnmarshalValueListPacked is an alternative to the original UnmarshalValueList
+// The difference is that the original provided measurements like (PLUGIN_DSNAME, ["value",xxx])
+// while this one will provide measurements like (PLUGIN, {["DSNAME",xxx]})
+// this effectively joins collectd data that should go together, such as:
+//  (df, {["used",1000],["free",2500]})
+func (s *Service) UnmarshalValueListPacked(vl *api.ValueList) []models.Point {
+	timestamp := vl.Time.UTC()
+
+	var name = vl.Identifier.Plugin
+	tags := make(map[string]string)
+	fields := make(map[string]interface{})
```

Comment:
Although Go allows you to `make` maps (and slices) without specifying a size, but we've run into many bottlenecks that were (at least partially) addressed by specifying an initial size hint. 

Here, I'd go with 
```
tags := make(map[string]string, 4)
fields := make(map[string]interface{}, len(vl.Values))
```
We might not use all 4 `tags` slots, which is fine, but giving `fields` the right capacity from the start will avoid creating some garbage as the map is grown from a smaller capacity.